mtags/src/main/scala/scala/meta/internal/metals/Docstrings.scala

```diff
@@ -138,6 +140,29 @@ class Docstrings(index: GlobalSymbolIndex) {
     }
   }
 
+  private def maybeCacheAlternative(
+      defn: SymbolDefinition,
+      contentType: ContentType
+  ) = {
+    val defSymbol = defn.definitionSymbol.value
+    val querySymbol = defn.querySymbol.value
+    lazy val queryContent = Content.from(querySymbol, contentType)
+
+    if (defSymbol != querySymbol && !cache.contains(queryContent)) {
+      cache
+        .get(Content.from(defSymbol, contentType))
+        .foreach(cache.put(queryContent, _))
+
+      alternativesMap.synchronized {
```

Comment:
Are we actually getting anything from alternativesMap? Looks like we only cache info there

Reply:
Look at line `195`, it tells us which symbols to remove from the cache, so we remove also the alternative symbols.

Reply:
Would it make sense to put the alternatives to MetalsSymbolDocumentation instead?

Reply:
If `a/filenname$package/O.` is an alternative for `a/O.` and `docs` are docs for `a/O.`.

I put into cache:
   - `a/O.` -> `docs`
   - `a/filenname$package/O.` -> `docs`

So when `a/filenname$package/O.` comes, we find docs in ~ constant time.

And I put into `alternativesMap`:
  - `a/O.` -> `a/filenname$package/O.`, so when we unindexed `a/O.`, we also unindexed `a/filenname$package/O.`

And this mapping is also found in ~ constant time. If put into `MetalsSymbolDocumentation`, I have to iterate over whole cache to find this connection.

Reply:
Ach, right. I was hoping to avoid another map, but I guess we can't

Reply:
We can also not cache alternatives and assume this happens rarely enough that indexing the file every time for them is fine (that was the behaviour before).